lib/puppet/provider/certmonger_certificate/certmonger_certificate.rb

```diff
@@ -0,0 +1,149 @@
+Puppet::Type.type(:certmonger_certificate).provide :certmonger_certificate do
+  desc "Provider for certmonger certificates."
+
+  confine :exists => "/usr/sbin/certmonger"
+  commands :getcert => "/bin/ipa-getcert"
+
+  mk_resource_methods
+
+  def initialize(value={})
+    super(value)
+    @property_flush = {}
+  end
+
+  def create
+    @property_flush[:ensure] = :present
+  end
+
+  def exists?
+    @property_hash[:ensure] == :present
+  end
+
+  def destroy
+    @property_flush[:ensure] = :absent
+  end
+
+  def self.get_list_of_certs
+    output = getcert('list')
+    return parse_cert_list(output)
+  end
+
+  def self.parse_cert_list(list_output)
+    output_array = list_output.split("\n")
```

Comment:
It would be a new dependency but it would be more reliable to use the certmonger dbus interface to get this information. Some example code for the paths and objects can be found in https://git.fedorahosted.org/cgit/freeipa.git/tree/ipapython/certmonger.py